### DISCUSSION TITLE
🚀 Light mode 😎

### DISCUSSION THREAD

**Discussion by alex3305** [louislam/dockge#201] [Feature Request]
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

UI Feature

### 🔖 Feature description

I would really like a light mode. The dark mode is a bit too dark for me currently.

### ✔️ Solution

Same as with Uptime Kuma, that we can switch between light and dark modes.

**Comment by serafin-tech** [comment]
1. thanks for the work on Dockge - really useful tool
2. **bumping it up** - first thing which triggered big resistance in me for Dockge is this too dark mode. Maybe there is an UX expert here who can help with that?

**Comment by KenroE67** [comment]
I would be forever grateful for light mode. I have to use a browser extensions to invert the colors so I can even use it.

**Reply by wsw70** [reply]
> I have to use a browser extensions to invert the colors so I can even use it.

Which extension did you use for that?

**Reply by without-ordinary** [reply]
> > I have to use a browser extensions to invert the colors so I can even use it.
> 
> Which extension did you use for that?

I misuse DarkReader to make semi-functional light mode for sites that only have dark. It has issues doing this but at least sites are more accessible for me.

**Comment by andersmmg** [comment]
I personally use dark mode, but I know multiple people that struggle with reading with dark mode, especially those with impaired vision.

**Comment by CampaniaGuy** [comment]
Hi, i too have a problem with dark mode in dockge. It uses too dark colors, causing eye strain.
I noticed there already is a light-mode implemented that just needs to be shown in the settings menu.
From what i see, only terminal and editors are missing the light-mode. Are they the only components needing a change?

**Comment by wsw70** [comment]
This dark mode is very difficult for some people due to the choice of colors. I have issues with some colors but with dockge I do not even see where the entry fields are.

For reference I use the default vscode dark mode and it is more or less fine (it is at the limit though).

**Comment by jmarmstrong1207** [comment]
@louislam I was thinking of making the dark mode a bit lighter with more dark gray color palette, which is what @wsw70 was referencing. How would you think of that? From my experience, absolute dark mode can be difficult to use when there's any light glare on the screen, requiring a ton of contrast. Lightening the dark mode will make it more universal :)

This is a very rough estimate of what it may look like
![image](https://github.com/user-attachments/assets/4655d772-32e3-4719-8730-459a0fcbd4f5)

**Reply by KenroE67** [reply]
OMG, Yes. I have such a hard time with the current dark mode.

**Reply by andrewrdavidson** [reply]
I'd really like a proper light mode but this would be a big improvement.

**Reply by CampaniaGuy** [reply]
I may prefer a slightly darker gray but that would be an improvement nonetheless so go ahead and star a PR when you compete it. It would be a great contribution to me.

**Reply by lubarb** [reply]
I just installed dockge. With my monitor and setup, at first I didn't even realize that the screens had different gray and black sections, so they made no sense. Please, please make the gray lighter at least as an option. As is, dockge is unusable for me.

**Comment by SootyOwl** [comment]
Hi all,

I have had success enabling the existing-but-disabled theme setting via [Stylus](https://github.com/openstyles/stylus).

To do so, create a new style via the extension for the `<baseurl>/settings/appearance` page and paste this snippet in:

```
/* Enable the theme setting in Settings > Appearance */
.my-4 {
	display: inherit !important;
}
```

Then save it and navigate to Settings > Appearance in Dockge to be greeted by the Theme option:
![image](https://github.com/user-attachments/assets/ae5e9c23-30f8-4707-9951-c26b58014289)

(Thanks to https://github.com/louislam/dockge/pull/575 for showing that the option already exists, it's just disabled.)


**EDIT:** I've just [made this available on userstyles.world](https://userstyles.world/style/23025/dockge-enable-theme-setting) to simplify the installation process, you can just hit "Install" and then edit the domain for your purposes (assuming you have Stylus already).

**Reply by without-ordinary** [reply]
Thanks for the workaround. I can see why this was disable, the light mode has some issues, but at least I can use this. xD

Dark mode only is an accessibility blocker for me, white text on black causes pulsating ghosting which leads nausea and migraines after about a minute.

**Reply by misudo** [reply]
No light mode to date? Very disappointing, but I'm sure there are higher priorities right now. Very much looking forward to proper light mode feature, please! 🙏
